ee/tables/systemprofiler/systemprofiler.go

```diff
@@ -102,16 +102,20 @@ func (t *Table) generate(ctx context.Context, queryContext table.QueryContext) (
 
 	datatypeQ, ok := queryContext.Constraints["datatype"]
 	if !ok || len(datatypeQ.Constraints) == 0 {
-		return results, fmt.Errorf("The %s table requires that you specify a constraint for datatype", t.tableName)
+		return results, fmt.Errorf("the %s table requires that you specify a constraint for datatype", t.tableName)
+	}
+
+	// Check for maximum number of datatypes
+	if len(datatypeQ.Constraints) > 3 {
+		return results, fmt.Errorf("maximum of 3 datatypes allowed per query, got %d", len(datatypeQ.Constraints))
 	}
 
 	for _, datatypeConstraint := range datatypeQ.Constraints {
 		dt := datatypeConstraint.Expression
 
-		// If the constraint is the magic "%", it's eqivlent to an `all` style
+		// Block the % wildcard
 		if dt == "%" {
-			requestedDatatypes = []string{}
-			break
+			return results, fmt.Errorf("wildcard %% is not allowed as a datatype constraint")
```

Comment:
small nit: errors.New and only a single `%`

Larger question: We often _don't_ return errors from tables. If we do, it can kill the whole query. Instead we often log the error, and `continue`. I'm a little agnostic here, but it's probably safer to `continue`

Reply:
I moved the allowed characters to be handled during the GetConstraints and kept the returning error only in the scenario where we have more datatypes than the limit